src/runtime/generators.js

```diff
@@ -52,6 +52,31 @@
                       state);
   }
 
+  // The following unique object serves as a non-catchable exception raised by
+  // the implementation of Generator.prototype.return
+  var SENTINEL = new Object();
+  
+  function GeneratorWrapper(iterator) {
+    this.iterator = iterator;
+  }
+  GeneratorWrapper.prototype = {
+    next: function (v) {
+      return this.iterator.next(v);
+    },
+    throw: function (e) {
+      if (e === SENTINEL) {
+        if (this.iterator.return) {
+          this.iterator.return(e.value);
+        }
+        throw e;
```

Comment:
Yes, I have to end any dependent generator (via the call to `this.iterator.return`) and then the generator itself (via `throw e`). This is in http://people.mozilla.org/~jorendorff/es6-draft.html#sec-generator-function-definitions-runtime-semantics-evaluation.

But after re-reading the spec again, I think I have missed something, which I will have to rectify: Under "YieldExpression : yield * AssignmentExpression", 1.c.x in the cited section, the `returnValue` becomes the one of the closed dependent  generator.